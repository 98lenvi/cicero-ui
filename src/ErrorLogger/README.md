## Cicero-UI - ErrorLogger

### Usage

```
npm install @accordproject/cicero-ui
```

```
import { ErrorLogger } from '@accordproject/cicero-ui';

const ErrorContainer = props => (
  <ErrorComponent errors={props.errors}/>
);
```

### Props

- `error` : An `array` which contains error objects.
