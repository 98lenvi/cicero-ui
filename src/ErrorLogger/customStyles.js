import styled from 'styled-components';

export const CustomStylesWrapper = styled.div`
.errorHeader{
};
.errorDisplay{
};
.errorBarArrow{
};
.errorComponent{
};
.errorFile{
};
.errorArrowDiv{
};
.errorType{
};
.errorShortMessage{
};
.errorFullMessage{
};
`