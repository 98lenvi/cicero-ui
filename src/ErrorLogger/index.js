import React from 'react';
import PropTypes from 'prop-types';
// import ReactTable from 'react-table';
import styled from 'styled-components';
import 'react-table/react-table.css';

const ErrorsBar = styled.div`
  width: 100vw;
  height: 100px;
  padding: 10px;
  background-color: #1E2D53;
  box-shadow: 0 -2px 20px 0 rgba(20,31,60,0.65);
`;

const ErrorsHeader = styled.div`
  height: 20px;
  width: 66px;
  color: ${props => (props.errors ? '#c71a19' : '#19c6c7')};
  transition: 0.5s;
  font-family: "IBM Plex Sans";
  font-size: 16px;
  font-weight: bold;
  letter-spacing: -0.5px;
  line-height: 20px;
`;

// /**
//  * @param {object} d
//  * @param {string} key
//  */
// function buildMessage(d, key) {
//   let result = 'Unknown';

//   if (d.fileLocation) {
//     result = '';
//     if (d.fileLocation[key].line) {
//       result += `Line: ${d.fileLocation[key].line}`;
//     }

//     if (d.fileLocation[key].column) {
//       result += ` Col: ${d.fileLocation[key].column}`;
//     }
//   }

//   return result;
// }

// /**
//  * @param {object} d
//  */
// function buildStartLocation(d) {
//   return buildMessage(d, 'start');
// }

// /**
//  * @param {object} d
//  */
// function buildEndLocation(d) {
//   return buildMessage(d, 'end');
// }

const errorsExist = err => err.length > 0;

const errorsLength = err => err.length || 'No';


// const columns = [{
//   Header: 'Type',
//   accessor: 'type',
//   width: 100,
// },
// {
//   Header: 'Name',
//   accessor: 'name',
//   width: 200,
// },
// {
//   Header: 'File',
//   accessor: 'fileName',
//   width: 100,
// },
// {
//   id: 'startLocation',
//   Header: 'Start Location',
//   accessor: d => buildStartLocation(d),
//   width: 100,
// },
// {
//   id: 'endLocation',
//   Header: 'End Location',
//   accessor: d => buildEndLocation(d),
//   width: 100,
// },
// {
//   Header: 'Message',
//   accessor: 'shortMessage',
//   width: 600,
// },
// ];

const ErrorLogger = (props) => {
  console.log('props: ', props.errors);
  return (
  <div>
    {/* <ReactTable
      data={props.errors}
      pageSize={5}
      columns={columns}
    /> */}
    <ErrorsBar>
      <ErrorsHeader errors={errorsExist(props.errors)}>
        {errorsLength(props.errors)} Errors
      </ErrorsHeader>
    </ErrorsBar>
  </div>
  );
};

ErrorLogger.propTypes = {
  errors: PropTypes.array.isRequired,
};

export default ErrorLogger;
