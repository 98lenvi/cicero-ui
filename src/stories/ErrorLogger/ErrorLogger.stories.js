import React from "react";
import docs from "../../ErrorLogger/README.md";
import { withA11y } from "@storybook/addon-a11y";
import ErrorLogger from "../../ErrorLogger";
import { withKnobs, text } from "@storybook/addon-knobs";

export default { title: "'Components/Error Logger" };

export const errorLogger = () => {
  const errors = {
    error_1: {
      modelError: "Some Error"
    }
  };
  const errorsPropsObject = {
    ERRORS_HEADER_BACKGROUND: text("ERRORS_HEADER_BACKGROUND", ""),
    ERRORS_HEADER_BACKGROUND_HOVER: text("ERRORS_HEADER_BACKGROUND_HOVER", ""),
    ERRORS_HEADER_EXPAND_ARROW: text("ERRORS_HEADER_EXPAND_ARROW", ""),
    ERRORS_HEADER_BORDER_TOP: text("ERRORS_HEADER_BORDER_TOP", ""),
    ERRORS_HEADER_SHADOW: text("ERRORS_HEADER_SHADOW", ""),
    ERRORS_DISPLAY_BACKGROUND: text("ERRORS_DISPLAY_BACKGROUND", ""),
    ERRORS_DISPLAY_SHADOW: text("ERRORS_DISPLAY_SHADOW", ""),
    ERRORS_DISPLAY_Z_INDEX: text("ERRORS_DISPLAY_Z_INDEX", ""),
    ERROR_BORDER_BOTTOM: text("ERROR_BORDER_BOTTOM", ""),
    ERROR_EXPAND_ARROW: text("ERROR_EXPAND_ARROW", ""),
    ERROR_FILE: text("ERROR_FILE", ""),
    ERROR_FILE_HOVER: text("ERROR_FILE_HOVER", ""),
    ERROR_TYPE: text("ERROR_TYPE", ""),
    ERROR_FULL_MESSAGE: text("ERROR_FULL_MESSAGE", ""),
    ERROR_SHORT_MESSAGE: text("ERROR_SHORT_MESSAGE", "")
  };
  const mockNavigateToClauseError = () => alert('Insert your navigation function here.')
  return (
    <ErrorLogger
      errors={errors}
      errorNav={mockNavigateToClauseError}
      errorsProps={errorsPropsObject}
    />
  );
};
errorLogger.story = {
  component: errorLogger,
  decorators: [withA11y, withKnobs],

  parameters: {
    notes: docs
  }
};
