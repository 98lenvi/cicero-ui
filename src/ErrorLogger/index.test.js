import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ErrorLogger from './index';

describe('<ErrorLogger />', () => {
  describe('on initialization', () => {
    it('renders page correctly', () => {
      const component = shallow(<ErrorLogger errors={[]} />);
      const tree = toJson(component);
      expect(tree).toMatchSnapshot();
    });
  });
});
