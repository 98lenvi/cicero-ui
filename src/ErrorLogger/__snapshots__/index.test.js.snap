// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorLogger /> on initialization renders page correctly 1`] = `
<div>
  <styled.div>
    <styled.div
      errors={false}
    >
      No
       Errors
    </styled.div>
  </styled.div>
</div>
`;
